src/components/ui/slider/AppSliderButtons.vue

```diff
@@ -49,14 +50,76 @@ export default defineComponent({
       )[0] as HTMLElement;
     });
 
-    const { showRightButton, showLeftButton, navigateNext, navigatePrevious } = useSliderButtons(
-      containerElement,
-    );
+    const {
+      clientWidth,
+      visibleAreaStart,
+      visibleAreaEnd,
+      canScrollLeft,
+      canScrollRight,
+    } = useScrollState(containerElement);
+
+    const navigateTo = (items: HTMLElement[], element: HTMLElement) => {
+      const firstItemToShowIndex = items.indexOf(element);
+      const firstItemToShowStart = element.offsetLeft;
+
+      const lastItemToShow = items.find((item, index) => {
+        if (index < firstItemToShowIndex) {
+          return false;
+        }
+        if (index === items.length - 1) {
+          return true;
+        }
+        const nextElementSibling = item.nextSibling as HTMLElement;
+        return (
+          nextElementSibling.offsetLeft + nextElementSibling.clientWidth >
+          firstItemToShowStart + clientWidth.value
+        );
+      });
+
+      const lastItemToShowEnd = lastItemToShow.offsetLeft + lastItemToShow.clientWidth;
+
+      const offset = -(clientWidth.value + firstItemToShowStart - lastItemToShowEnd) / 2;
+
+      VueScrollTo.scrollTo(element, 300, {
+        container: containerElement.value,
+        offset,
+        x: true,
+        y: false,
+      });
+    };
+
+    const getItems = (): HTMLElement[] => {
+      return [...containerElement.value.children] as HTMLElement[];
+    };
+
+    const navigatePrevious = () => {
+      const items = getItems();
+      const firstVisibleLeft = items.find((item) => item.offsetLeft >= visibleAreaStart.value);
+      const firstItemToShow = items.find(
+        (item) => item.offsetLeft + clientWidth.value >= firstVisibleLeft.offsetLeft,
+      );
+      if (firstItemToShow != null) {
+        navigateTo(items, firstItemToShow);
+      }
+    };
+
+    const navigateNext = () => {
+      const items = getItems();
+
+      const firstItemToShow = items.find(
+        (item) => item.offsetLeft + item.clientWidth >= visibleAreaEnd.value,
+      );
+
+      if (firstItemToShow != null) {
+        navigateTo(items, firstItemToShow);
+      }
+    };
+
```

Comment:
This logic was really coupled to this component, so I moved it back from its function.